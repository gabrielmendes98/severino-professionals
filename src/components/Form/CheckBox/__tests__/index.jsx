import {
  FormikWrapper,
  toMatchSnapshot,
  render,
  screen,
  userEvent,
  fireEvent,
  waitFor,
} from 'test-utils';
import Checkbox from '..';

it('should render checkbox without errors', () => {
  toMatchSnapshot(
    <FormikWrapper initialValues={{ name: false }}>
      <Checkbox name="test" label="Test" />
    </FormikWrapper>,
  );
});

it('should render checkbox with errors', async () => {
  render(
    <FormikWrapper
      initialValues={{ test: false }}
      validate={() => ({ test: 'Required Field' })}
      onSubmit={() => {}}
    >
      <Checkbox name="test" label="Test" />
      <button type="submit">submit</button>
    </FormikWrapper>,
  );

  userEvent.click(screen.getByRole('button', { name: /submit/i }));
  expect(await screen.findByText(/required field/i)).toBeInTheDocument();
});

it('should set value when click on checkbox', () => {
  render(
    <FormikWrapper initialValues={{ test: false }}>
      <Checkbox name="test" label="Test" />
    </FormikWrapper>,
  );

  const checkBox = screen.getByLabelText(/test/i);
  expect(checkBox).not.toBeChecked();
  userEvent.click(checkBox);
  expect(checkBox).toBeChecked();
});
