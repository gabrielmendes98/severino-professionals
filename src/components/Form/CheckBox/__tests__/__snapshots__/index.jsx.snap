// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render checkbox without errors 1`] = `
<DocumentFragment>
  <form
    action="#"
  >
    <div
      class="MuiFormControl-root"
    >
      <legend
        class="MuiFormLabel-root"
      />
      <div
        class="MuiFormGroup-root"
      >
        <label
          class="MuiFormControlLabel-root"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-4"
                data-indeterminate="false"
                name="test"
                type="checkbox"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Test
          </span>
        </label>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
